### PR TITLE
Overhaul Javadoc of RestTemplateBuilder

### PR DESCRIPTION
**Pull request by aahlenst**
While working on `RestTemplateBuilder`, I noticed a couple of typos, some confusing explanations, a lack of descriptions of error conditions, and that it does not always follow the Spring Framework style guide. So I tried my best to improve it.

If you're not happy with some choices I made, feel free to edit it yourself.

### CODE REVIEW

File: src/main/java/org/springframework/retry/support/RetryTemplateBuilder.java [hunk 1]
```diff
@@ -497,10 +506,11 @@ public RetryTemplateBuilder withListener(RetryListener listener) {
 	}
 
 	/**
-	 * Appends all provided {@code listeners} to {@link RetryTemplate}'s listener list.
+	 * Appends all provided {@code listeners} to {@link RetryTemplate}'s list of
```
Comment (artembilan) [inline src/main/java/org/springframework/retry/support/RetryTemplateBuilder.java:509]: The method Javadoc must be imperative: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting

I see that in some place you have it fixed, but not here.

**Review by artembilan — CHANGES_REQUESTED** [review]
Just great!

Please, update the Copyright of the affected classes to the current year and add your name to the `@author` list.

Thanks

**Comment by aahlenst** [comment]
Problems should be fixed. As far as I can see, I'm already mentioned as an`@author` on all the classes.

**Comment by artembilan** [comment]
@aahlenst ,

thank you for contribution; looking forward for more!

**Comment by xak2000** [comment]
Small correction. This task is not about `RestTemplateBuilder` as stated in the title, but about `RetryTemplateBuilder`. :)

Noticed "Overhaul Javadoc of RestTemplateBuilder" in v2.0.2 release notes and was very surprised. 😄

**Comment by artembilan** [comment]
Fixed.
Thank you! 😄

**Pull request merged**